src/client/Microsoft.Identity.Client/Internal/Requests/Interactive/AuthCodeExchangeComponent.cs

```diff
@@ -22,28 +23,28 @@ internal class AuthCodeExchangeComponent : ITokenRequestComponent
         private readonly string _authorizationCode;
         private readonly string _pkceCodeVerifier;
         private readonly TokenClient _tokenClient;
+        private readonly string _clientInfo;
 
         public AuthCodeExchangeComponent( 
             AuthenticationRequestParameters requestParams,
             AcquireTokenInteractiveParameters interactiveParameters,
             string authorizationCode,
-            string pkceCodeVerifier)
+            string pkceCodeVerifier,
+            string clientInfo)
         {
             _requestParams = requestParams ?? throw new ArgumentNullException(nameof(requestParams));
             _interactiveParameters = interactiveParameters ?? throw new ArgumentNullException(nameof(interactiveParameters));
             _authorizationCode = authorizationCode ?? throw new ArgumentNullException(nameof(authorizationCode));
             _pkceCodeVerifier = pkceCodeVerifier ?? throw new ArgumentNullException(nameof(pkceCodeVerifier));
+            _clientInfo = clientInfo;
 
```

Comment:
should we throw here if it's null? otherwise, if using ClientInfo, we will throw later. how often will this be null from the /authorize endpoint? do we know in which cases it will be null, and if so, can we send an emtpy string or something else to avoid an issue?
_clientInfo = clientInfo ?? string.Empty

Reply:
If no client info is sent from the authorize endpoint and this value is null MSAL will not send any CCS routing info. We are only using the client info here for CCS which is not mandatory for the token exchange so it is ok if we don't send any routing headers when it is not available. No need to throw in this case. 

I'm not sure if ADFS or B2C will send the client info but I do not believe those technologies will have CCS support.

Reply:
i understand that, but i'm saying to use the ClientInfo class that we already have, and that class will throw if it's null, so that's something to work around, but I would use that class because it has the correct error handling and test coverage. the other methods do not. Should see if other authorization servers return client info from the /authorize endpoint. And in which cases will this be null, if ever. I don't think it's common for it to be null from the /token endpoint.

Reply:
I have this code here which will prevent the ClientInfo class from constructing the object and throwing the exception. 
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/cb136556765bbccb285b2ae737e3a337dae9b477/src/client/Microsoft.Identity.Client/Internal/Requests/Interactive/AuthCodeExchangeComponent.cs#L66

I figured this will workaround that issue.

Reply:
i think you should still use the clientInfo class because it has the correct error handle for json issues.

Reply:
I am using it here
https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/cb136556765bbccb285b2ae737e3a337dae9b477/src/client/Microsoft.Identity.Client/Internal/Requests/Interactive/AuthCodeExchangeComponent.cs#L64-L83

If the client info is null then I don't create the client info class but if it is not null, then it is fed into the `ClientInfo.CreateFromJsonClientInfo.CreateFromJson` If there is a JSON issue it will be caught.

Reply:
I think there are some missing words or typos in your last comment. can you please rephrase why I should remove the method?